RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -411,6 +411,41 @@ class PumpOpsSynchronous {
         }
         return frameData
     }
+
+    internal func readPumpStatus() throws -> PumpStatus {
+        let clockResp: ReadTimeCarelinkMessageBody = try getMessageBodyWithType(.ReadTime)
+
+        let pumpModel = try getPumpModel()
+
+        let resResp: ReadRemainingInsulinMessageBody = try getMessageBodyWithType(.ReadRemainingInsulin)
+
+        let reservoir = resResp.getUnitsRemainingForStrokes(pumpModel.strokesPerUnit)
+
+        let battResp: GetBatteryCarelinkMessageBody = try getMessageBodyWithType(.GetBattery)
+
+        let statusResp: ReadPumpStatusMessageBody = try getMessageBodyWithType(.ReadPumpStatus)
+
+        return PumpStatus(clock: clockResp.dateComponents, batteryVolts: battResp.volts, suspended: statusResp.suspended, bolusing: statusResp.bolusing, reservoir: reservoir, model: pumpModel)
```

Comment:
Did you say the battery status had the quarter-percentage encoded as well? Should we return that too?